src/ir/possible-contents.cpp

```diff
@@ -887,6 +887,14 @@ struct InfoCollector
     }
     addRoot(curr, PossibleContents::exactType(curr->type));
   }
+  void visitArrayNewSeg(ArrayNewSeg* curr) {
+    if (curr->type == Type::unreachable) {
+      return;
+    }
+    // TODO: How to add a connection between data/elem segments and the
+    // allocated value / its type?
+    WASM_UNREACHABLE("What is this supposed to do???");
```

Comment:
@kripken, guidance on what to do here would be much appreciated.

Reply:
As the segments are not immutable I guess the values here must be roots. So something like
```cpp
addRoot(DataLocation{type, 0}, PossibleContents::exactType(type from the segment?));
```